runtime/src/iree/hal/drivers/hip/hip_device.c

```diff
@@ -1031,64 +1379,81 @@ static iree_status_t iree_hal_hip_device_queue_alloca(
     iree_hal_allocator_pool_t pool, iree_hal_buffer_params_t params,
     iree_device_size_t allocation_size,
     iree_hal_buffer_t** IREE_RESTRICT out_buffer) {
+  *out_buffer = NULL;
+  IREE_TRACE_ZONE_BEGIN(z0);
+
   iree_hal_hip_device_t* device = iree_hal_hip_device_cast(base_device);
-  IREE_RETURN_IF_ERROR(
-      iree_hal_hip_set_context(device->hip_symbols, device->hip_context));
+  uint64_t queue_affinity_mask =
```

Comment:
I need to add some helpers for iree_hal_queue_affinity_t so we don't have to cast to uint64_t (this is fine for now, just noting - I think I have some helpers in my branch as I ran into the same case)